.docker/Dockerfile

```diff
@@ -1,23 +1,109 @@
 ARG ROS_DISTRO=rolling
-FROM ros:$ROS_DISTRO-ros-base AS ci
+FROM ros:$ROS_DISTRO-ros-base AS base
+
+# Create the non-root user early because both "mavros" and "robot"
+# stages depend its existence
+RUN apt-get -q update \
+    && apt-get -q -y upgrade \
+    && apt-get -q install --no-install-recommends -y \
+    git \
+    gosu \
```

Comment:
I'm open to leaving gosu in here if we split this into its own temporary Dockerfile, but if we add it to the original, we should consider a separate PR.